### DISCUSSION TITLE
Increase max brightness for 2021 Macbook Pros

### DISCUSSION THREAD

**Discussion by sagz** [MonitorControl/MonitorControl#815] [Ideas]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Is your feature request related to a problem? Please describe

Principally, this would be useful for working outdoors in sunlight.

Push the brightness of the new XDR display macbook pros past the default allowance of 500 nits. 
I want to be able to push the display to the advertised "sustained 1000 nits" every once in a while for short durations when doing a video call outdoors.

### Describe the solution you'd like

Apple advertises the new Mini-LED MacBook Pro 14 with a peak brightness of up to 1600 nits and a sustained brightness of 1000 nits, but these values seem to be limited to HDR contents. Standard SDR contents are [limited to 500 nits](https://www.notebookcheck.net/The-new-MacBook-Pro-14-only-manages-500-nits-with-SDR-contents.575327.0.html), which is comparable to the current MacBook Pro 13.

### Describe alternatives you've considered

I'm wondering if it's possible to hack around this somehow.

Is it possible to simulate HDR content as an overlay across the screen? It's a dirtier hack for sure, but might work (there are reports that with HDR windows, MacOS makes only those windows in HDR while other SDR windows stay in lower brightness)

Another hack I wonder is if one could create a 'custom preset' from such a list:
![image](https://user-images.githubusercontent.com/964843/139787202-0c95fb4f-fd9c-4139-a939-e53d37be2aa7.png)

**Comment by waydabber** [comment]
I think it might be possible but I have no MBP and even don't have a HDR display to experiment with (only a humble 13" MBA with 8Gigs of RAM and a trusted old SDR display lol). :)

**Comment by MacInTheNet** [comment]
If you can solve the problem I think all the community that is discussing this around the world (on macrumors for instance) would happily buy you whatever you need XD

**Comment by namork** [comment]
as a photographer i would be very happy to use a brighter macbook display outdoors. where/how can i support the development of this feature?

**Comment by waydabber** [comment]
well I don't think I can work on that without an actual MacBook Pro and even if I had one there is no guarantee so the contribution is not guaranteed to yield any return. It would be unreasonable to ask ppl to contribute that much for this feature and based on the average rate of donations, it will take about 3-4 more years for the project to gather enough money so I could buy one from my share. I have some ideas but this would involve heavy experimentation to see what works and what doesn't.

But surely there are other coders who have some experience in this department with a new MBP. I'll add a "Help Wanted" badge to this issue, hopefully that will attract someone to start working on this from the community. If by some miracle I can lay my hands on a new MBP sometime in the future and the problem is not solved yet, of course I'll try whatever (silly or not so silly - who knows) ideas I have regarding making this happen.

**Comment by waydabber** [comment]
(but hey if you are a photographer and use HDR images, the nMBP display should already show them with higher brightness in supported apps).

**Comment by namork** [comment]
dear waydrabber. i work in lightroom and the user interface would be the most important thing to make readable better in the sunlight.

**Comment by waydabber** [comment]
I have a way of achieving this now at least in principle. I plan to add this to [BetterDummy](https://github.com/waydabber/BetterDummy) as it has the core technologies (screen virtualization) that is needed (short version: a virtualized screen will be streamed to a persistent full screen view with custom metal tone mapping layer to extend the dynamic range into HDR/EDR headroom beyond SDR - the result should be an SDR screen that can crank up the max brightness to full HDR peak brightness - and probably kill the display on the long run if not used wisely lol).

Of course the principle might not work itself into reality or not easily (but the streaming part is done and works splendidly).

**Reply by chipweinberger** [reply]
This would be great to see a video of. If you share the code, I'm happy to test it on my 14" that arrives soon.

**Reply by waydabber** [reply]
Well, I delved into it a bit more and the feature is not as simple to achieve as I hope, the app would need to manually tone-map every frame of the virtual display stream to HDR. It might take a bit more time to achieve this and it's not my first priority at the moment, but it's not forgotten. :)

**Comment by waydabber** [comment]
I'll convert this into a discussion since this probably won't make into MonitorControl.

**Reply by waydabber** [reply]
https://github.com/waydabber/BetterDummy/issues/163

**Comment by Amachik** [comment]
hi, there is tool for that. it is called Vivid.. it is day and night. 

https://www.getvivid.app/privacy

ps.. it is paid. but it is worth the money.

**Reply by waydabber** [reply]
Hey guys, I recommend [Lunar](https://lunar.fyi) by @alin23 for this purpose - it is actually a much more feature complete app than Vivid with much more features while not significantly more costly. Also it's XDR brightness implementation is superior to Vivid as it does not use overlays which I think Vivid uses (I didn't investigate so I might not be right about this - MonitorControl also uses overlays for dimming as well but only as a last resort since it has drawbacks).

Once I get there I'll implement this as there are now multiple methods to achieve XDR brightening plus thanks to the donations from MonitorControl and [BetterDummy](https://github.com/waydabber/BetterDummy) I now have a 14" MacBook Pro as well. Summer is coming and I'll probably use mine outdoors as well so that extra brightness might be useful *. :)

_(*) hopefully it won't make the panel explode..._

**Reply by alin23** [reply]
Thanks @waydabber for spreading the word about Lunar!

For people interested in the **XDR Brightness** feature, there's also a $10-off coupon giveaway on Twitter this week for a Lunar Pro license, so you'll be able to buy Lunar for even less than Vivid's price: [Twitter coupon giveaway](https://twitter.com/alinp32/status/1511060832716824581)

I've tested Vivid and it's a nice little tool, but its approach has some drawbacks which I've outlined in this [Reddit post](https://www.reddit.com/r/apple/comments/tvzzsn/vivid_force_hdr_mode_on_your_xdr_dislplay_macbook/i3gu1mk/)

**Reply by waydabber** [reply]
I'll be implementing this here for now: https://github.com/waydabber/BetterDummy/issues/499

**Comment by waydabber** [comment]
note: there is a [beta version of BetterDummy](https://github.com/waydabber/BetterDummy/releases/tag/v1.2.0-beta) that has this feature. It should work both for XDR and any other HDR display. I use a new technique which do not emply overlays and undocumented API calls + autotunes the output based on the available brightness range reported by the system so it should work with all displays without need for special calibration. 

That's the theory of course, I was able to test this on my own 14" MPB and my Lenovo HDR display only.

**Reply by waydabber** [reply]
+ I also shared how this works with @alin23 and he is planning to upgrade Lunar with the newer method. I also gained some insights from his implementation so it was a mutually beneficial exchange of ideas.
